lib/ruby/signature/prototype/rb.rb

```diff
@@ -94,13 +94,22 @@ def process(node, namespace:, current_module:, comments:)
             source_decls.push mod
 
             each_node module_body do |child|
-              process child, namespace: mod.name.to_namespace, current_module: mod, comments: comments
+              process child, namespace: mod.name.to_namespace, current_module: mod, comments: comments, singleton: false
+            end
+          when :SCLASS
+            this = node.children[0]
+            if this.type != :SELF
+              Signature.logger.warn "`class <<` syntax with not-self may be compiled to incorrect code: #{this}"
```

Comment:
The generator will generate incorrect rbs for the following ruby code.


```ruby
class << Foobar
  def foo() end
end
```


I think it is a super rare case, so it just warns about it.